samples/core/Miscellaneous/NewInEFCore7/DocumentsContext.cs

```diff
@@ -0,0 +1,595 @@
+        PublicationDate = publicationDate;
+        CoverArt = coverArt;
+    }
+
+    public int Id { get; private set; }
+
+    [Timestamp]
+    public byte[]? RowVersion { get; set; }
+
+    public string Title { get; set; }
+    public int NumberOfPages { get; set; }
+    public DateTime PublicationDate { get; set; }
+    public byte[]? CoverArt { get; set; }
+
+    public DateTime FirstRecordedOn { get; private set; }
+    public DateTime RetrievedOn { get; private set; }
+}
+
+public class Book : Document
+{
+    public Book(string title, int numberOfPages, DateTime publicationDate, byte[]? coverArt)
+        : base(title, numberOfPages, publicationDate, coverArt)
+    {
+    }
+
+    public string? Isbn { get; set; }
+
+    public List<Person> Authors { get; } = new();
+}
+
+public class Magazine : Document
+{
+    public Magazine(string title, int numberOfPages, DateTime publicationDate, byte[]? coverArt, int issueNumber)
+        : base(title, numberOfPages, publicationDate, coverArt)
+    {
+        IssueNumber = issueNumber;
+    }
+
+    public int IssueNumber { get; set; }
+    public decimal? CoverPrice { get; set; }
+    public Person Editor { get; set; } = null!;
+}
+
+public class Person
+{
+    public Person(string name)
+    {
+        Name = name;
+    }
+
+    public int Id { get; private set; }
+
+    [ConcurrencyCheck]
+    public string Name { get; set; }
+
+    public ContactDetails Contact { get; set; } = null!;
+
+    public List<Book> PublishedWorks { get; } = new();
+    public List<Magazine> Edited { get; } = new();
+}
+
+public abstract class DocumentsContext : DbContext
+{
+    public bool LoggingEnabled { get; set; }
+    public abstract MappingStrategy MappingStrategy { get; }
+    public virtual bool UseStoredProcedures => true;
+
+    public DbSet<Document> Documents => Set<Document>();
+    public DbSet<Book> Books => Set<Book>();
+    public DbSet<Magazine> Magazines => Set<Magazine>();
+    public DbSet<Person> People => Set<Person>();
+
+    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        => optionsBuilder.UseSqlServer(@$"Server=(localdb)\mssqllocaldb;Database={GetType().Name}")
+            .EnableSensitiveDataLogging()
+            .LogTo(
+                s =>
+                {
+                    if (LoggingEnabled)
+                    {
+                        Console.WriteLine(s);
+                    }
+                }, LogLevel.Information);
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        modelBuilder.Entity<Document>(
+            entityTypeBuilder =>
+            {
+                entityTypeBuilder.Property(document => document.FirstRecordedOn).HasDefaultValueSql("getutcdate()");
+                entityTypeBuilder.Property(document => document.RetrievedOn).HasComputedColumnSql("getutcdate()");
+
+                if (UseStoredProcedures)
+                {
+                    switch (MappingStrategy)
```

Comment:
Since this is a sample I think it would be easier to understand if each mapping strategy was handled in a different method (perhaps on the derived context)